### PR TITLE
OCPBUGS-36390: aws: do not require create permissions when BYO IAM role 

### PR DESCRIPTION
**Pull request by r4f4**
The Installer is unconditionally requiring permissions needed to create IAM roles even when the users use existing roles. They should be included only when needed.

This PR kind of reverts https://github.com/openshift/installer/pull/5286. IAM roles created by the Installer are now consistently tagged with "owned". We should also tag BYO roles so we know which clusters are using them, and so that it's
not deleted by the installer during cluster destroy.

### CODE REVIEW

**Comment by r4f4** [comment]
Testing this locally with an existing IAM role, I see during destroy: 
```
DEBUG Search for and remove shared tags for IAM roles matching kubernetes.io/cluster/rdossant-installer-07-hsc8w: shared
DEBUG iterating over a page of 100 IAM roles
DEBUG iterating over a page of 52 IAM roles
DEBUG Removing the shared tag from the "rdossant-installer-master-role" IAM role
DEBUG Removing the shared tag from the "rdossant-installer-worker-role" IAM role
```
and the BYO roles are not deleted.

**Comment by r4f4** [comment]
/cc @barbacbd
Since you were involved in related efforts like https://github.com/openshift/installer/pull/7537

**Comment by r4f4** [comment]
Note for reviewers: the existing pre-submit CI jobs won't check the permissions set for the BYO role case is enough. Hopefully they'll be enough to check for regressions in the case the Installer creates the roles.

File: pkg/asset/cluster/aws/aws.go [hunk 1]
```diff
@@ -95,6 +97,64 @@ func tagSharedVPCResources(ctx context.Context, clusterID string, installConfig
...
+	session, err := installConfig.AWS.Session(ctx)
+	if err != nil {
+		return fmt.Errorf("could not create AWS session: %w", err)
+	}
+
+	tagKey, tagValue := sharedTag(clusterID)
+
+	iamClient := iam.New(session, aws.NewConfig().WithRegion(installConfig.Config.Platform.AWS.Region))
+	for role := range iamRoles {
+		if _, err := iamClient.TagRoleWithContext(ctx, &iam.TagRoleInput{
```
Comment (barbacbd) [inline pkg/asset/cluster/aws/aws.go:145]: nit: should we make a child context outside of the loop and pass that ?

Comment (r4f4) [reply pkg/asset/cluster/aws/aws.go:145]: I think we discussed this off band and I forgot to add the update here. Contrary to GCP, in the AWS code we don't do much fine-tuning of contexts. So I'll leave this as is for now, and we can do the fine-tuning as its own effort.

**Comment by r4f4** [comment]
/hold

I think [IncludeCreateInstanceRole](https://github.com/openshift/installer/pull/8688/commits/9f35e4995db34166f9b265c86abbbaa949f6d0ba#diff-548e72871741032796b39d4e800c78905a07e79d0a08a0de53a34e645bead268R36) might not be completely correct. I'll investigate and add unit tests.

**Comment by r4f4** [comment]
> During cluster deletion should we be removing the shared tag added to the BYO IAM role? IMHO, we should but not aware of the convention so far.

[We should](https://github.com/openshift/installer/blob/master/pkg/destroy/aws/aws.go#L226-L229) and that's what happened when I tested it. The code is generic and not dependent on resource types. That's why we have the include the `iam:UntagRole` permission when using BYO role.

**Comment by r4f4** [comment]
Update: move permission groups generation to a function so it can be unit tested and add unit tests.

**Comment by patrickdillon** [comment]
/approve

this lgtm but want to go over it more carefully before labelling. Anyone else should feel free to label, or i will after another pass

**Comment by r4f4** [comment]
unit tests will probably fail until https://github.com/openshift/installer/pull/8762 merges.

**Pull request merged**